### PR TITLE
Cert health check can use duration syntax for deadline param

### PR DESCRIPTION
**Pull request by zachmargolis**
- Ex: /api/health/certs?deadline=7d

**Why**: This allows us to query the endpoint with fixed strings
for separate durations, so we could warn at a 30d expiration but
have a separate check that errors at a 7d expiration

Notes: I cribbed the `DurationParser` class I wrote in the IDP. It didn't seem worth forcing into a shared gem or anything, for reference, it's at: https://github.com/18F/identity-idp/blob/master/app/services/duration_parser.rb

### CODE REVIEW

File: .reek.yml [hunk 1]
```diff
@@ -1,31 +0,0 @@
----
```
Comment (zachmargolis) [inline .reek.yml:1]: We had already disabled reek in #165, but this file stuck around. While tracking down the method complexity error CodeClimate error I found this and figured I'd remove it. Can do another PR if anybody feels strongly

File: app/services/duration_parser.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  # @param value [String, nil]
+  def initialize(value)
+    @value = value
+  end
+
+  # @return [ActiveSupport::Duration, nil]
+  def parse
+    return if value.blank?
+
+    match = value.match(/^(?<number>\d+)(?<duration>\D)$/)
```
Comment (aduth) [inline app/services/duration_parser.rb:14]: I guess the switch fallthrough would have the same effect of generating a `nil` value, but... should we match on the specific duration characters we support?

```suggestion
    match = value.match(/^(?<number>\d+)(?<duration>[dwmy])$/)
```

File: app/services/duration_parser.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  def parse_duration(number, duration)
+    case duration
+    when 'd' # days
+      number.days
+    when 'w' # weeks
+      (7 * number).days
+    when 'm' # months
+      (30 * number).days
+    when 'y' # years
+      (365 * number).days
```
Comment (aduth) [inline app/services/duration_parser.rb:36]: Are we okay that these aren't strictly accurate? e.g. months with fewer or more than 30 days, or leap years.

Would it work to just...
```suggestion
    when 'w' # weeks
      number.weeks
    when 'm' # months
      number.months
    when 'y' # years
      number.years
```

**Review by aduth — APPROVED** [review]

File: app/services/duration_parser.rb (hunk 3 above)
Comment (zachmargolis) [reply app/services/duration_parser.rb:36]: When originally implementing this I decided to make it simpler/dumber so I could easily document it:

See `verified_within` (inside the details dropdown) on https://developers.login.gov/oidc/

The ActiveSupport methods are very nice but they use fractional quantities for some of those so it seemed more predictable this way (so when the fractional quantities add up they don't change hours/minutes as much)

```ruby
1.years.to_i.to_f / 1.day.to_i
=> 365.2425
```

I feel somewhat strongly about leaving it simpler and documented, but if you feel strongly, I could change it here and discuss changing the IDP with the rest of the team

File: app/services/duration_parser.rb (hunk 2 above)
Comment (zachmargolis) [reply app/services/duration_parser.rb:14]: My thinking was the switch fallthrough would be the easiest? If we ever add a new character we only need to add it on one place, also allows like multicharacter signifiers if we ever wanted to allow like `hr` and `wk` or something

File: app/services/duration_parser.rb (hunk 3 above)
Comment (aduth) [reply app/services/duration_parser.rb:36]: > I feel somewhat strongly about leaving it simpler and documented, but if you feel strongly, I could change it here and discuss changing the IDP with the rest of the team

I don't feel strongly.

👍

File: app/services/duration_parser.rb (hunk 2 above)
Comment (aduth) [reply app/services/duration_parser.rb:14]: > My thinking was the switch fallthrough would be the easiest? If we ever add a new character we only need to add it on one place, also allows like multicharacter signifiers if we ever wanted to allow like `hr` and `wk` or something

Seems fine to me. And I just now noticed you have test coverage for this with the `123x` example, so should be protected against accidental regressions. On that point, I guess my primary concern is not that the switch fallthrough doesn't work, just that it might not be as obvious to the next person that we're relying on it.
